racket/src/cs/c/main.c

```diff
@@ -53,6 +53,7 @@ static void scheme_set_dll_procs(scheme_dll_open_proc open,
 # define embedded_dll_close NULL
 #endif
 
+__attribute__((used))
```

Comment:
I think we'll need the usual macro approach to include this `__attribute__` only for compilers that support it.

Reply:
Yes - we need to do that. I haven't checked yet - should the headers where we have `MZ_` stuff like this be shared with Chez code? Or do you want to have a new macro on the Chez side for this?

Reply:
We already have this definition under `PRESERVE_IN_EXECUTABLE` in `start/config.inc` so I will just use that.

Reply:
It will have to be a new macro. It's not even the Chez side, exactly, since "src/cs/c/main.c" is outside of Chez Scheme proper (so, it doesn't see the misnamed "version.h", for example, where some macros like this live).

Reply:
Great!